### PR TITLE
Adding function to decide if HDNode is private

### PR DESCRIPTION
**Pull request by runn1ng**
It's a very simple function that probably isn't that important, but can help someone.

If you like this idea, I will add some tests for that

### CODE REVIEW

**Comment by dcousens** [comment]
Nice!

Maybe `isNeutered`?
That way it is consistent with the `neutered()` interface?

**Comment by dcousens** [comment]
Also, are you up to add some tests? :+1:

**Comment by runn1ng** [comment]
Hm. On one hand, it makes sense, because it's consistent, on the other hand I just wouldn't guess from the name what it means.

I will look more into the travis failures (seems like style issue... that's what I get for editing code in github GUI :))

File: src/hdnode.js [hunk 1]
```diff
@@ -281,6 +281,10 @@ HDNode.prototype.deriveHardened = function (index) {
   return this.derive(index + HDNode.HIGHEST_BIT)
 }
 
+HDNode.prototype.isPrivate = function () {
+  return !!(this.keyPair.d);
```
Comment (fanatid) [inline src/hdnode.js]: extra semicolon

**Comment by dcousens** [comment]
why not both?

I wouldn't be opposed to having both then I guess?  Its just more to maintain and think about.
I'm just wanting to be consistent, especially since `isPrivate` is equally vague,  at least the `isNeutered` introduces less context.

@fanatid thoughts?

**Comment by dcousens** [comment]
In any case @runn1ng,  please add tests in lieu of the bike shed.

**Comment by fanatid** [comment]
Since `HDNode` already has [neutered](https://github.com/bitcoinjs/bitcoinjs-lib/blob/daafb9794e60a7329c7d5825fceab71e17f6f0a7/src/hdnode.js#L147), `isNeutered` is better.

**Comment by runn1ng** [comment]
I have renamed the function and removed the semicolon. Now I am adding tests

**Comment by runn1ng** [comment]
I have added some tests. Is it OK like this?

**Comment by runn1ng** [comment]
...stupid semicolons! :D ok now?

**Comment by runn1ng** [comment]
....the test failure is timeout failure and not related to my changes (I think).

File: test/hdnode.js [hunk 2]
```diff
@@ -240,7 +242,7 @@ describe('HDNode', function () {
     var f = fixtures.valid[0]
 
     it('strips all private information', function () {
-      var hd = HDNode.fromBase58(f.master.base58, NETWORKS_LIST)
+      var hd = HDNode.fromBase58(f.master.base58Priv, NETWORKS_LIST)
```
Comment (dcousens) [inline test/hdnode.js]: ha.  Good catch.

File: src/hdnode.js [hunk 3]
```diff
@@ -239,7 +239,7 @@ HDNode.prototype.derive = function (index) {
 
   // Private parent key -> private child key
   var derivedKeyPair
-  if (this.keyPair.d) {
+  if (!this.isNeutered()) {
```
Comment (dcousens) [inline src/hdnode.js]: I'm not sure if `isNeutered` is that great after all.
Its hard to derive the intended meaning... `isPrivate` would be much clearer here.

@runn1ng could you add both?
We'll know by `3.0.0` as to whether we should clean it up or not...

**Comment by dcousens** [comment]
Sorry for the screw around @runn1ng... :+1:

**Comment by dcousens** [comment]
Bugger it actually, thanks @runn1ng :+1: 

If `.isNeuetered` ends up being a bit too weird or confusing, we'll use `isPrivate`.

**Comment by dcousens** [comment]
Won't publish for a little while yet,  will be in `2.3.0`

**Pull request merged**